core/src/main/java/com/wttech/aem/acm/core/assist/Assistancer.java

```diff
@@ -13,62 +13,30 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import org.apache.sling.api.resource.ResourceResolver;
-import org.osgi.framework.BundleContext;
-import org.osgi.framework.BundleListener;
-import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Deactivate;
-import org.osgi.service.component.annotations.Modified;
 import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Component(immediate = true, service = Assistancer.class)
 public class Assistancer {
 
+    private static final Logger LOG = LoggerFactory.getLogger(Assistancer.class);
+
     @Reference
     private transient OsgiScanner osgiScanner;
 
     @Reference
     private transient ResourceScanner resourceScanner;
 
-    private BundleContext bundleContext;
-
-    private BundleListener bundleListener;
-
     private List<ClassInfo> classCache = Collections.emptyList();
 
-    private int bundlesHashCode;
-
-    @Activate
-    protected void activate(BundleContext bundleContext) {
-        this.bundleContext = bundleContext;
-        this.bundleListener = event -> updateCache();
```

Comment:
IMO that was wrong; imagine instance startup or restart; any bundles installed after ours will trigger redundant cache update